### PR TITLE
🎉 Source S3: Reduce image size and add acceptance test

### PR DESCRIPTION
**Pull request by flash1293**
Adding the experimental document type parser bloated the image size unnecessarily. This PR reduces the required dependencies.

It also adds an acceptance test to verify the parser works correctly for regular PDFs and OCR-only PDFs (e.g. scans)

⚠️ This PR means that the `hi_res` parsing strategy can't be applied anymore - it will default to `fast` and fall back to `ocr_only` if necessary. ⚠️

### CODE REVIEW

**Comment by airbyte-oss-build-runner** [comment]
##  source-s3 test report (commit [`a971df111d`](https://github.com/airbytehq/airbyte/pull/31654/commits/a971df111dc872c21f6c4760204be5322ad42201)) - ❌

⏲️  Total pipeline duration: 15mn10s 

| Step                                                      | Result   |
|:----------------------------------------------------------|:---------|
| Build source-s3 docker image for platform(s) linux/x86_64 | ✅       |
| Unit tests                                                | ✅       |
| Acceptance tests                                          | ❌       |
| Check our base image is used                              | ✅       |
| Code format checks                                        | ✅       |
| Validate metadata for source-s3                           | ✅       |
| Connector version semver check                            | ✅       |
| Connector version increment check                         | ✅       |
| QA checks                                                 | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_s3-improve-and-test/1697804043/a971df111dc872c21f6c4760204be5322ad42201/source-s3/4.1.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:a971df111dc872c21f6c4760204be5322ad42201)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-s3 test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-s3 test report (commit [`e76c488b42`](https://github.com/airbytehq/airbyte/pull/31654/commits/e76c488b42edf8e9d673599a53245e2fe90fc953)) - ❌

⏲️  Total pipeline duration: 14mn35s 

| Step                                                      | Result   |
|:----------------------------------------------------------|:---------|
| Build source-s3 docker image for platform(s) linux/x86_64 | ✅       |
| Unit tests                                                | ✅       |
| Acceptance tests                                          | ❌       |
| Check our base image is used                              | ✅       |
| Code format checks                                        | ✅       |
| Validate metadata for source-s3                           | ✅       |
| Connector version semver check                            | ✅       |
| Connector version increment check                         | ✅       |
| QA checks                                                 | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_s3-improve-and-test/1697806521/e76c488b42edf8e9d673599a53245e2fe90fc953/source-s3/4.1.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:e76c488b42edf8e9d673599a53245e2fe90fc953)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-s3 test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-s3 test report (commit [`eb6941d301`](https://github.com/airbytehq/airbyte/pull/31654/commits/eb6941d301800ffcc8a644a32fbfdd31f0236ced)) - ✅

⏲️  Total pipeline duration: 06mn19s 

| Step                                                      | Result   |
|:----------------------------------------------------------|:---------|
| Build source-s3 docker image for platform(s) linux/x86_64 | ✅       |
| Unit tests                                                | ✅       |
| Acceptance tests                                          | ✅       |
| Check our base image is used                              | ✅       |
| Code format checks                                        | ✅       |
| Validate metadata for source-s3                           | ✅       |
| Connector version semver check                            | ✅       |
| Connector version increment check                         | ✅       |
| QA checks                                                 | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_s3-improve-and-test/1697807750/eb6941d301800ffcc8a644a32fbfdd31f0236ced/source-s3/4.1.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:eb6941d301800ffcc8a644a32fbfdd31f0236ced)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-s3 test
```

**Review by bazarnov — APPROVED** [review]

**Comment by flash1293** [comment]
One other thing I need to check here is whether we should pull in pytesseract

**Comment by airbyte-oss-build-runner** [comment]
##  source-s3 test report (commit [`f628f5831e`](https://github.com/airbytehq/airbyte/pull/31654/commits/f628f5831e7c0781b26379be2830114fb10f529a)) - ❌

⏲️  Total pipeline duration: 14mn29s 

| Step                                                     | Result   |
|:---------------------------------------------------------|:---------|
| Build source-s3 docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                               | ✅       |
| Acceptance tests                                         | ✅       |
| Check our base image is used                             | ✅       |
| Code format checks                                       | ❌       |
| Validate metadata for source-s3                          | ✅       |
| Connector version semver check                           | ✅       |
| Connector version increment check                        | ✅       |
| QA checks                                                | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_s3-improve-and-test/1698059517/f628f5831e7c0781b26379be2830114fb10f529a/source-s3/4.1.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:f628f5831e7c0781b26379be2830114fb10f529a)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-s3 test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-s3 test report (commit [`254aa91d6d`](https://github.com/airbytehq/airbyte/pull/31654/commits/254aa91d6d0f568ed0136305c1b8ce021ff2508b)) - ✅

⏲️  Total pipeline duration: 14mn38s 

| Step                                                     | Result   |
|:---------------------------------------------------------|:---------|
| Build source-s3 docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                               | ✅       |
| Acceptance tests                                         | ✅       |
| Check our base image is used                             | ✅       |
| Code format checks                                       | ✅       |
| Validate metadata for source-s3                          | ✅       |
| Connector version semver check                           | ✅       |
| Connector version increment check                        | ✅       |
| QA checks                                                | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_s3-improve-and-test/1698060831/254aa91d6d0f568ed0136305c1b8ce021ff2508b/source-s3/4.1.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:254aa91d6d0f568ed0136305c1b8ce021ff2508b)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-s3 test
```

File: airbyte-integrations/connectors/source-s3/build_customization.py [hunk 1]
```diff
@@ -28,4 +28,5 @@ async def post_connector_install(connector_container: Container) -> Container:
         connector_container.with_new_file("/tmp/nltk_python_script.py", nltk_python_script)
         .with_exec(["python", "/tmp/nltk_python_script.py"], skip_entrypoint=True)
         .with_exec(["rm", "/tmp/nltk_python_script.py"], skip_entrypoint=True)
+        .with_exec(["sh", "-c", "apt-get update && apt-get install -y tesseract-ocr poppler-utils"], skip_entrypoint=True)
```
Comment (alafanechere) [inline airbyte-integrations/connectors/source-s3/build_customization.py:31]: This is not related to `nltk` so it might make the doctrsing less accurate.
Would you mind splitting `post_connector_install` into two sub functions like "def seed_nltk_data" and `def install_tesseract` (and explain the role of these system deps in the docstring 🙏 .

File: airbyte-integrations/connectors/source-s3/source_s3/v4/unstructured_parser.py [hunk 2]
```diff
@@ -74,7 +74,16 @@ def read_file(self, file_handle: IOBase, logger: logging.Logger) -> str:
             return optional_decode(file_handle.read())
         if filetype not in self._supported_file_types():
             raise RecordParseError(FileBasedSourceError.ERROR_PARSING_RECORD, filename=file_name)
-        elements = partition(file=file_handle, metadata_filename=file_name)
+
+        if filetype == FileType.PDF:
```
Comment (alafanechere) [inline airbyte-integrations/connectors/source-s3/source_s3/v4/unstructured_parser.py:78]:
```suggestion
        if filetype is FileType.PDF:
```

**Review by alafanechere — APPROVED** [review]
👍 Just a suggestion to make the `build_customization.py` module more readable / documented.

**Review by alafanechere — CHANGES_REQUESTED** [review]

**Comment by airbyte-oss-build-runner** [comment]
##  source-s3 test report (commit [`efa0727cb8`](https://github.com/airbytehq/airbyte/pull/31654/commits/efa0727cb8a94c3a72865519b020a6e97705252c)) - ✅

⏲️  Total pipeline duration: 13mn32s 

| Step                                                     | Result   |
|:---------------------------------------------------------|:---------|
| Build source-s3 docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                               | ✅       |
| Acceptance tests                                         | ✅       |
| Check our base image is used                             | ✅       |
| Code format checks                                       | ✅       |
| Validate metadata for source-s3                          | ✅       |
| Connector version semver check                           | ✅       |
| Connector version increment check                        | ✅       |
| QA checks                                                | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_s3-improve-and-test/1698069256/efa0727cb8a94c3a72865519b020a6e97705252c/source-s3/4.1.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:efa0727cb8a94c3a72865519b020a6e97705252c)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-s3 test
```

**Comment by flash1293** [comment]
@alafanechere thanks, fixed!

**Comment by airbyte-oss-build-runner** [comment]
##  source-s3 test report (commit [`1eadd5df96`](https://github.com/airbytehq/airbyte/pull/31654/commits/1eadd5df968ec55926f771d50c481e1af3aef22d)) - ✅

⏲️  Total pipeline duration: 13mn40s 

| Step                                                     | Result   |
|:---------------------------------------------------------|:---------|
| Build source-s3 docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                               | ✅       |
| Acceptance tests                                         | ✅       |
| Check our base image is used                             | ✅       |
| Code format checks                                       | ✅       |
| Validate metadata for source-s3                          | ✅       |
| Connector version semver check                           | ✅       |
| Connector version increment check                        | ✅       |
| QA checks                                                | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_s3-improve-and-test/1698070418/1eadd5df968ec55926f771d50c481e1af3aef22d/source-s3/4.1.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:1eadd5df968ec55926f771d50c481e1af3aef22d)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-s3 test
```

**Review by alafanechere — APPROVED** [review]

**Review by aaronsteers — APPROVED** [review]
LGTM!

**Comment by airbyte-oss-build-runner** [comment]
##  source-s3 test report (commit [`f601bef3d7`](https://github.com/airbytehq/airbyte/pull/31654/commits/f601bef3d778837ef15d43ead6900baf22f50058)) - ✅

⏲️  Total pipeline duration: 16mn06s 

| Step                                                     | Result   |
|:---------------------------------------------------------|:---------|
| Build source-s3 docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                               | ✅       |
| Acceptance tests                                         | ✅       |
| Check our base image is used                             | ✅       |
| Code format checks                                       | ✅       |
| Validate metadata for source-s3                          | ✅       |
| Connector version semver check                           | ✅       |
| Connector version increment check                        | ✅       |
| QA checks                                                | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_s3-improve-and-test/1698229640/f601bef3d778837ef15d43ead6900baf22f50058/source-s3/4.1.3/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:f601bef3d778837ef15d43ead6900baf22f50058)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-s3 test
```
